src/util/RtreeBuild.cpp

```diff
@@ -0,0 +1,1133 @@
+     * This is used when a leaf node is reached.
+   */
+  if (this->GetOrderedBoxes().WorkInRam()) {
+    for(RTreeValueWithOrderIndex box : *this->GetOrderedBoxes().GetRectanglesInRam()) {
+      Node leafNode = Node(box.id, box.box);
+      this->AddChild(leafNode);
+    }
+  } else {
+    FileReader fileReader = FileReader(this->GetOrderedBoxes().GetRectanglesOnDisk());
+
+    std::optional<RTreeValueWithOrderIndex> element = fileReader.GetNextElement();
+    while(element) {
+      Node leafNode = Node(element.value().id, element.value().box);
+      this->AddChild(leafNode);
+      element = fileReader.GetNextElement();
+    }
+
+    fileReader.Close();
+  }
+}
+
+OrderedBoxes ConstructionNode::GetOrderedBoxes() {
+  return this->orderedBoxes;
+}
+
+void Node::AddChild(Node& child) {
+  Rtree::BoundingBox box = child.GetBoundingBox();
+  uint64_t entryId = child.GetId();
+  RTreeValue entry = {box, entryId};
+  this->children.push_back(entry);
+}
+
+Rtree::BoundingBox Node::GetBoundingBox() const {
+  return this->boundingBox;
+}
+
+void Node::SetIsLastInnerNode(bool isLast) {
+  this->isLastInnerNode = isLast;
+}
+
+uint64_t Rtree::SaveNode(Node &node, bool isLastInnerNode, std::ofstream& nodesOfs) {
+  node.SetIsLastInnerNode(isLastInnerNode);
+
+  uint64_t pos = static_cast<uint64_t>(nodesOfs.tellp());
+  boost::archive::binary_oarchive archive(nodesOfs);
+  archive << node;
+  nodesOfs.write(" ", 1);
+
+  return pos;
+}
+
+std::optional<Rtree::BoundingBox> GetBoundingBoxFromWKT(const std::string& wkt) {
+  /**
+     * Parse the wkt literal in a way, that only the relevant data for the rtree gets read in.
+   */
+  bool lookingForX = true;
+  bool readingDouble = false;
+  std::string currentDouble;
+
+  double minX = -1;
+  double maxX = -1;
+  double minY = -1;
+  double maxY = -1;
+
+  for (char c : wkt) {
+    if (isdigit(c)) {
+      readingDouble = true;
+      currentDouble += c;
+    } else if (c == '.') {
+      readingDouble = true;
+      currentDouble += '.';
+    } else if (c == ' ') {
+      if (readingDouble && lookingForX) {
+        // x is completely read in
+        readingDouble = false;
+        lookingForX = false;
+        double x;
+        try {
+          x = std::stod(currentDouble);
+        } catch(...) {
+          return { };
+        }
+        currentDouble = "";
+        if (x < minX || minX == -1) {
+          minX = x;
+        }
+
+        if (x > maxX) {
+          maxX = x;
+        }
+      }
+    } else {
+      if (readingDouble && !lookingForX) {
+        // y is completely read in
+        readingDouble = false;
+        lookingForX = true;
+        double y;
+        try {
+          y = std::stod(currentDouble);
+        } catch(...) {
+          return { };
+        }
+        currentDouble = "";
+        if (y < minY || minY == -1) {
+          minY = y;
+        }
+
+        if (y > maxY) {
+          maxY = y;
+        }
+      }
+    }
+  }
+
```

Comment:
As discussed, check the performance of a much smaller and cleaner implementations via the `CTRE` library (compile-time-regular-expressions).